ui/lib/kv/addon/components/page/secret/details.hbs

```diff
@@ -2,65 +2,77 @@
   <:tabLinks>
     <LinkTo @route="secret.details" data-test-secrets-tab="Secret">Secret</LinkTo>
     <LinkTo @route="secret.metadata.index" data-test-secrets-tab="Metadata">Metadata</LinkTo>
-    <LinkTo @route="secret.metadata.versions" data-test-secrets-tab="Version History">Version History</LinkTo>
-    <LinkTo @route="secret.metadata.diff" data-test-secrets-tab="Version Diff">Version Diff</LinkTo>
+    {{#unless @noMetadataPermission}}
```

Comment:
There are different types of metadata permissions. Technically, if we've returned a 403 from a query then it's `noMetadataListPermissions`. If we got a 403 from a queryRecord it's `noMetadataReadPermission`. I'm seeing in the old code, we do something similar to what you have by using a model param,`failedServerRead`. But I wonder if we can be more specific by using `noMetadataReadPermissions` here. Permissions are super tricky to keep track of so when we can I'd like to make it obvious what kind of permissions they don't have.

Reply:
I realize that the handleResponse is catching all failures for the adapter, but if there is a way for us to understand which method failed? Or maybe we only expect this to happen on queryRecord?

Reply:
Yeah after sleeping on it over the weekend I realized this needs to be more explicit. I'll change this and instead grab it from the `kv/data` model!